core/engine_test.go

```diff
@@ -968,3 +968,71 @@ func TestMinIterationDuration(t *testing.T) {
 	// But we expect the custom counter to be added to 4 times
 	assert.Equal(t, 4.0, getMetricSum(collector, "testcounter"))
 }
+
+//nolint: funlen
+func TestMinIterationDurationInSetupTeardownStage(t *testing.T) {
+	t.Parallel()
+	setupScript := `
+		import { sleep } from "k6";
+
+		export function setup() {
+			sleep(1);
+		}
+
+		export let options = {
+			minIterationDuration: "2s",
+			duration: "2s",
+			setupTimeout: "2s",
+		};
+
+		export default function () {
+		};`
+	teardownScript := `
+		import { sleep } from "k6";
+
+		export let options = {
+			minIterationDuration: "2s",
+			duration: "2s",
+			teardownTimeout: "2s",
+		};
+
+		export default function () {
+		};
+
+		export function teardown() {
+			sleep(1);
+		}
+`
+	tests := []struct {
+		name, script string
+	}{
+		{"Test setup", setupScript},
+		{"Test teardown", teardownScript},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			runner, err := js.New(
+				&loader.SourceData{URL: &url.URL{Path: "/script.js"}, Data: []byte(tc.script)},
+				nil,
+				lib.RuntimeOptions{},
+			)
+			require.NoError(t, err)
+			engine, err := NewEngine(local.New(runner), runner.GetOptions())
+			require.NoError(t, err)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			errC := make(chan error)
+			go func() { errC <- engine.Run(ctx) }()
+			select {
+			case <-time.After(10 * time.Second):
+				cancel()
```

Comment:
put the cancel in a defer ?

Reply:
Have no idea, I carry the test code from other case and change the script for setup/teardown only.